source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -700,7 +742,10 @@ The event API here is assumed to be like the standard `Python Event
             serverSupportsStreaming = description.type != Unknown and description.topologyVersion != Null
             connectionIsStreaming = connection != Null and connection.moreToCome
             transitionedWithNetworkError = isNetworkError(description.error) and previousDescription.type != Unknown
-            if serverSupportsStreaming or connectionIsStreaming or transitionedWithNetworkError:
+            if streamingEnabled and serverSupportsStreaming and not rttMonitor.started:
+                # Start the RttMonitor.
+                rttMonitor.run()
```

Comment:
Is there a reason we want to disable the RTT monitor for servers that don't support streaming heartbeat (i.e. server versions before 4.4)?

From reading the rationale "Why disable the streaming protocol on FaaS platforms like AWS Lambda?", it sounds like the goal of not starting the RTT monitor is to reduce open connections for FaaS environments. However, the suggested change would disable the RTT monitor for all databases earlier than 4.4, not just FaaS environments.

Reply:
This isn't a new requirement - drivers shouldn't be creating RTT monitors when the polling protocol is used.  The rtt monitors are only necessary when using the awaitable hello protocol, because then a round trip to the server can take up to `maxAwaitTimeMS`.  https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#measuring-rtt

Reply:
The spec wording and pseudo-code don't match your assertion.

The [Measuring RTT](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#measuring-rtt) section specifically says drivers can still use a separate RTT monitor for any server version:
> For consistency, clients MAY use dedicated connections to measure RTT for all servers, even those that do not support awaitable hello or legacy hello.

The current pseudo-code from the [Monitor thread](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#monitor-thread) section also describes always starting the RTT monitor, independent of server version:
```python
class Monitor(Thread):
...

  def run():
      # Start the RttMonitor.
      rttMonitor.run()
...
```

The new spec wording in this PR only suggests disabling the RTT monitor when streaming is _disabled_, but nothing about whether or not it's _supported_:
> When the streaming protocol is disabled the client MUST use the polling protocol and MUST NOT start an extra thread or connection for Measuring RTT.

Disabling the RTT monitor when streaming isn't supported is allowed by the spec. However, putting it in the pseudo-code makes it seem like a new requirement that conflicts with the spec wording.

Reply:
@ShaneHarvey Can you opine on this?

Reply:
Yes this PR changes the rules to disallow using an extra thread/socket for RTT on <4.4 servers. The rationale is that the extra connection limits scaling and puts unnecessary load on the server, especially in Lambda where MongoClients are not shared. Although the motivation is mostly Lambda this is still a beneficial change for all environments.

Reply:
To @matthewdale 's point, should this section from  [Measuring RTT](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#measuring-rtt) be removed or updated to reflect these rule changes? :

> For consistency, clients MAY use dedicated connections to measure RTT for all servers, even those that do not support awaitable hello or legacy hello.

Reply:
Good point. I updated this section to say clients MUST NOT dedicated connections to measure RTT in this case.

Reply:
Thanks for the clarification. Does that imply that the polling monitor should measure RTT (e.g. to support CSOT)? For the Go driver, that would significantly increases the scope of this change because we'd have to build RTT monitoring capability into the polling monitor.

Reply:
I'm confused by this question because the polling protocol always included RTT monitoring.

Reply:
Looking closer, the actual change in the Go driver would probably be be minimal. It does add complexity because there is no longer a single source of RTT data for CSOT, but the change only impacts a few lines of code.